zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/QueryResolver.scala

```diff
@@ -30,20 +30,26 @@ object QueryResolver {
   }
 
   private def resolveQuery(query: UnresolvedQuery, schemaMap: Map[String, Schema]): Query = {
-    val schema = schemaMap(query.dataset)
-    val fieldMap = schema.fieldMap
-    val (append, fieldMapAfterAppend) = resolveAppends(query.append, fieldMap)
-    val (lookup, fieldMapAfterLookup) = resolveLookups(query.lookup, fieldMapAfterAppend, schemaMap)
-    val (unnest, fieldMapAfterUnnest) = resolveUnnests(query.unnest, fieldMapAfterLookup)
-
-    val (filter, fieldMapAfterFilter) = resolveFilters(query.filter, fieldMapAfterUnnest)
-
-    val (groups, fieldMapAfterGroup) = resolveGroup(query.groups, fieldMapAfterFilter, schemaMap)
-    val (select, fieldMapAfterSelect) = resolveSelect(query.select, fieldMapAfterGroup)
-
-    val (globalAggr, fieldMapAfterGlobalAggr) = resolveGlobalAggregate(query.globalAggr, fieldMapAfterSelect)
-
-    Query(query.dataset, append, lookup, filter, unnest, groups, select, globalAggr, query.estimable)
+    schemaMap(query.dataset) match {
+      case temporal: TemporalSchema =>
+        val fieldMap = temporal.fieldMap
+
+        val (append, fieldMapAfterAppend) = resolveAppends(query.append, fieldMap)
+        val (lookup, fieldMapAfterLookup) = resolveLookups(query.lookup, fieldMapAfterAppend, schemaMap)
+        val (unnest, fieldMapAfterUnnest) = resolveUnnests(query.unnest, fieldMapAfterLookup)
+        val (filter, fieldMapAfterFilter) = resolveFilters(query.filter, fieldMapAfterUnnest)
+        val (groups, fieldMapAfterGroup) = resolveGroup(query.groups, fieldMapAfterFilter, schemaMap)
+        val (select, fieldMapAfterSelect) = resolveSelect(query.select, fieldMapAfterGroup)
+        val (globalAggr, fieldMapAfterGlobalAggr) = resolveGlobalAggregate(query.globalAggr, fieldMapAfterSelect)
+
+        Query(query.dataset, append, lookup, filter, unnest, groups, select, globalAggr, query.estimable)
+
+      case static: StaticSchema =>
+        val fieldMap = static.fieldMap
```

Comment:
What you want to do is that you can lookup a static dataset, instead of perform a lookup query for a static dataset.
I suggest it's better to keep the query resolver as it is, and move such logic to QueryValidator (which is used to validate a query).